invoices/invoiceregistry.go

```diff
@@ -613,10 +614,16 @@ func (i *InvoiceRegistry) SettleHodlInvoice(preimage lntypes.Preimage) error {
 	log.Debugf("Invoice(%v): settled with preimage %v", hash,
 		invoice.Terms.PaymentPreimage)
 
-	i.notifyHodlSubscribers(HodlEvent{
-		Hash:     hash,
-		Preimage: &preimage,
-	})
+	for key, htlc := range invoice.Htlcs {
+		if htlc.State == channeldb.HtlcStateCancelled {
```

Comment:
check that state is settled instead?

Reply:
Why settled? The invoice is settled and we need to notify all links that hold htlcs paying to the invoice. Htlcs that were already canceled before, don't need to receive anything and are skipped.

Reply:
Seems more correct to notify that it is settled _only if_ the state is settled. The only remaining state looks like it is `HtlcStateAccepted`, is it expected that we would find the htlc in this state here?

Reply:
Changed to checking for settled only and added comment